lib/index.ts

```diff
@@ -340,27 +340,37 @@ const getDeviceId = (device: usb.Device): string => {
 	return `${device.busNumber}:${device.deviceAddress}`;
 };
 
-const safeReadFile = async (filename: string): Promise<Buffer | undefined> => {
-	try {
-		return await readFile(Path.join(__dirname, '..', 'blobs', filename));
-	} catch (e) {
-		// no data
-	}
-};
-
 const getFileBuffer = async (
 	device: usb.Device,
 	filename: string,
+	extraFolder?: string,
 ): Promise<Buffer | undefined> => {
-	const folder =
-		device.deviceDescriptor.idProduct === USB_PRODUCT_ID_BCM2711_BOOT
-			? 'cm4'
-			: 'raspberrypi';
-	const buffer = await safeReadFile(Path.join(folder, filename));
-	if (buffer === undefined) {
-		debug("Can't read file", filename);
+	try {
+		if (extraFolder) {
+			const extraBuffer = await readFile(Path.join(extraFolder, filename));
```

Comment:
I think I may have encouraged you to do the wrong thing here: if readFile throws, this will now return undefined but it would previously fall through to the non-extraFolder readFile call and that might return something. Is this correct?

Reply:
Indeed good catch, it should be two try catch and not one...